test/unit_test/rag/utils/test_s3_conn.py

```
@@ -63,3 +66,69 @@ def test_s3_health_returns_false_on_client_error(monkeypatch):
+
+    with pytest.raises(ClientError):
+        storage.remove_bucket("dataset-id")
+
+
+def test_s3_remove_bucket_reports_partial_delete_failures(monkeypatch):
+    storage, client, _ = _new_storage(monkeypatch, {})
+    paginator = client.get_paginator.return_value
+    paginator.paginate.return_value = [{"Contents": [{"Key": "blocked"}]}]
+    client.delete_objects.return_value = {"Errors": [{"Key": "blocked", "Code": "AccessDenied"}]}
+
+    with pytest.raises(RuntimeError, match="blocked.*AccessDenied"):
```

Ruff (0.16.3)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)